landing_page: Restructure bottom navigation.

eeshangarg opened the pull request:
We recently added a lot of new pages to our top navigation and restructured top navigation in general. This commit updates the footer to reflect the recent changes to our top navigation.

@timabbott FYI :) @alya The only thing different from the original spec you gave me is the order of the "Solutions" pages. I mirrored what we have in the top navigation.

Desktop screenshot:
<img width="657" alt="footer_desktop" src="https://user-images.githubusercontent.com/7251823/127202622-1f163e7c-92b4-42e3-83c0-fc598df3a918.PNG">

Mobile (iPhone) screenshot:
<img width="277" alt="footer_mobile_1" src="https://user-images.githubusercontent.com/7251823/127202870-dbfd7635-1313-46a7-88be-3c9118071b44.PNG">
<img width="271" alt="footer_mobile_2" src="https://user-images.githubusercontent.com/7251823/127202871-f9580225-e183-4994-b724-b857e524281c.PNG">

Tablet (iPad) screenshot:
<img width="570" alt="footer_ipad" src="https://user-images.githubusercontent.com/7251823/127203009-1635f2d9-cca3-4fa2-bf06-7eb064421e43.PNG">

alya commented:
1. Could we make the columns wider so that fewer lines wrap? I think the columns don't have to all be the same width; we can try making them just as wide as they need to be to avoid wrapping.
2. Let's try dropping "Resources" from the headings as you suggested.
3. I think need to make the text darker; I don't know in detail about contrast checkers, but it's quite light. It's definitely hard to read on http://zulipdev.com:9991/accounts/register/, which has a light grey background.

eeshangarg commented:
@alya: I addressed all your feedback, except on tablets, we cannot have variable widths since they messed up the alignment of the two vertical blocks quite a bit. Here are the latest screenshots:

Web:
<img width="797" alt="footer_desktop" src="https://user-images.githubusercontent.com/7251823/127372215-1ad6eb90-1322-4096-bb73-a15e662d42c3.png">

Register page that you linked:
<img width="765" alt="footer_register" src="https://user-images.githubusercontent.com/7251823/127372231-949268c6-a5a8-4f74-bcb2-c5c28b100c58.png">

iPad/Tablets:
<img width="563" alt="footer_ipad" src="https://user-images.githubusercontent.com/7251823/127372261-b82ca60a-386b-4c0a-8613-81d42ab7025d.png">

iPhone:
<img width="274" alt="footer_mobile_1" src="https://user-images.githubusercontent.com/7251823/127372281-2ee3c4cd-ffed-422b-8e05-1b563717c911.png">
<img width="279" alt="footer_mobile_2" src="https://user-images.githubusercontent.com/7251823/127372280-b9af2f29-4f53-40db-9079-bde0eb3d342b.png">

alya commented:
Thanks! Next round of feedback:

1. Let's make the layout be a grid, i.e. the second row of headings all at the same height.
2. We should add a bit more space above the headings; it looks a little cramped.
3. Let's bump the font size up a bit.
4. We should give this section its own background color, to create visual separation from the main content. It should be different from the white on our landing pages (e.g. /for/education) and the grey on the login/account creation pages (e.g. /login). Perhaps a light tan?

eeshangarg commented:
@alya I addressed all of your feedback and also added a different light tan background color. There are a couple of bugs on mobile as a result of my changes so I need to look into those. For now, here is what the desktop screenshot looks like:
<img width="768" alt="footer_desktop" src="https://user-images.githubusercontent.com/7251823/127569941-8427088c-c746-40f1-896b-2421d616dfa0.png">
<img width="999" alt="footer_register" src="https://user-images.githubusercontent.com/7251823/127570184-c5c40a7c-7434-4e26-ae03-7d16229440bc.png">

eeshangarg commented:
@alya: All of your feedback has now been addressed, including the point about ordering sections differently for mobile and tablets. Here are the latest screenshots:

Desktop:
<img width="757" alt="footer-desktop" src="https://user-images.githubusercontent.com/7251823/127696383-80fa7618-b72c-41e3-b32a-08d3c20b2ca8.png">

Mobile:
<img width="281" alt="footer-phone-1" src="https://user-images.githubusercontent.com/7251823/127696416-e45a6c18-65a5-4ba6-bc21-23d3f4c6371b.png">
<img width="280" alt="footer-phone-2" src="https://user-images.githubusercontent.com/7251823/127696417-d0bb93e5-5b61-4950-a8a0-2e8dd02c0d20.png">

Tablet:
<img width="573" alt="footer-ipad" src="https://user-images.githubusercontent.com/7251823/127696435-7c631faa-9a6d-4fe8-abb2-0989f7a56d8a.png">

eeshangarg commented:
@alya: Adjusted tablet section spacing. :)

Screenshot:
<img width="575" alt="footer-ipad" src="https://user-images.githubusercontent.com/7251823/127698907-a2aa89f2-e4e5-4830-a13d-57072778caa4.png">

eeshangarg commented:
@alya Added more vertical spacing to the top of the footer. Screenshot:

<img width="723" alt="vertical_spacing" src="https://user-images.githubusercontent.com/7251823/127703374-05a06f84-dddf-4bef-8bae-ae72e6008531.PNG">

eeshangarg commented:
Added more spacing on both the top and the bottom:

<img width="1148" alt="vertical_spacing" src="https://user-images.githubusercontent.com/7251823/127705932-81c66e83-4b8e-47dc-81cc-ec7fc6e91584.PNG">

eeshangarg commented:
@timabbott @alya: This is now ready for a final review. All of Alya's feedback has been addressed. Thanks! :)

timabbott commented:
OK I looked at the implementation and have a few comments:
* The footer should be implemented using CSS grid, rather than the complex template rearrangement -- I believe that will natively cause the right thing to happen at narrow widths with a lot less complexity.  We use it on the /for/education pages if you want an example, but it's also a topic with nice online tutorials.
* We played with this and everyone agreed we should make that bottom margin 50px, not 100px.  
* This should be its own commit, but the 100px/130px margins around the octopus on the landing page should also move to 50px, like so:
![image](https://user-images.githubusercontent.com/2746074/127719114-df0f95e9-cdab-44de-92be-f43cac100024.png)
* The "find account" link doesn't seem to appear in the "Account" section, is that expected?

eeshangarg commented:
@timabbott To respond to your comments (in order):
* I am looking into CSS grid layouts for this. If this is time-sensitive and I can't figure out a way to do this in the next couple of hours, we can also do the grid migration in a follow-up.
* Done.
* Done.
* The `find_team_link_disabled` is `True` locally so that's why the template doesn't show the "Find Accounts" link, so that is expected.

Thanks! :)

eeshangarg commented:
@timabbott Grid layout has now been implemented, you were right, it is a lot more elegant! :) @alya FYI latest screenshots are here. :)

Desktop:
<img width="861" alt="footer-desktop" src="https://user-images.githubusercontent.com/7251823/128085962-2592c2fd-2f0f-4993-9606-9bc191562a14.png">

Mobile:
<img width="284" alt="footer-phone-1" src="https://user-images.githubusercontent.com/7251823/128085988-80ca222e-eeda-4c4b-bf9f-e4901c8c0000.png">
<img width="282" alt="footer-phone-2" src="https://user-images.githubusercontent.com/7251823/128085986-17622231-036f-4d7e-b4e3-bcb5308999c7.png">

Tablet:
<img width="577" alt="footer-ipad" src="https://user-images.githubusercontent.com/7251823/128086034-e242a7ed-7b8b-47d8-80f5-2ad0b95aef47.png">

timabbott commented:
Merged, huge thanks for making this migration happen!
